src/compiler/scala/tools/nsc/typechecker/MacroAnnotationNamers.scala

```diff
@@ -0,0 +1,773 @@
+package scala.tools.nsc.typechecker
+
+// imported from scalamacros/paradise
+trait MacroAnnotationNamers { self: Analyzer =>
+  import global._
+  import analyzer._
+  import definitions._
+  import scala.reflect.internal.Flags._
+  import scala.reflect.internal.Mode._
+
+  override def newNamer(context: Context): Namer = new MacroAnnotationNamer(context)
+
+  private class MacroAnnotationNamer(context: Context) extends Namer(context) {
```

Comment:
Can we have this class (as well as the accompanying Typer) publicized? In our SemanticDB compiler plugin, we are hijacking namers and typers to retain some information about tree originals, so we'll need to inherit from this class to be compatible with macro annotations (https://github.com/scalameta/scalameta/blob/v3.7.3/semanticdb/scalac/library/src/main/scala-2.11.11/scala/meta/internal/semanticdb/scalac/SemanticdbAnalyzer.scala#L20).

Reply:
sure, but I am planning to rework this code for M5